### PR TITLE
Better preflight validation

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: .github/workflows/build-test-deploy.yaml [hunk 1]
```diff
@@ -56,7 +56,32 @@ jobs:
           path: bin/
       - uses: engineerd/setup-kind@v0.5.0
       - run: chmod +x bin/preflight
-      - run: ./bin/preflight --interactive=false --format=json https://preflight.replicated.com
+      - run: |
+          ./bin/preflight --interactive=false --format=json https://preflight.replicated.com > result.json
+          cat result.json
```
Comment (sgalsaleh) [inline .github/workflows/build-test-deploy.yaml:61]: why is this needed?

File: .github/workflows/build-test-deploy.yaml [hunk 2]
```diff
@@ -56,7 +56,32 @@ jobs:
...
+
+          EXIT_STATUS=0
+          if grep -q "was not collected" result.json; then
+            echo "Some files were not collected"
+            EXIT_STATUS=1
+          fi
+
+          if (( `jq '.pass | length' result.json` < 1 )); then
+            echo "No passing prelifhgts found"
```
Comment (sgalsaleh) [inline .github/workflows/build-test-deploy.yaml]:
```suggestion
            echo "No passing preflights found"
```

File: .github/workflows/build-test-deploy.yaml [hunk 3]
```diff
@@ -56,7 +56,32 @@ jobs:
...
+            EXIT_STATUS=1
+          fi
+
+          if (( `jq '.warn | length' result.json` < 1 )); then
+            echo "No warnings found"
+            EXIT_STATUS=1
+          fi
+
+          if (( `jq '.fail | length' result.json` < 1 )); then
+            echo "No failed prelifhgts found"
```
Comment (sgalsaleh) [inline .github/workflows/build-test-deploy.yaml]:
```suggestion
            echo "No failed preflights found"
```

File: .github/workflows/build-test-deploy.yaml (hunk 1 above)
Comment (divolgin) [reply .github/workflows/build-test-deploy.yaml:61]: so we can see what's in that file.  because this test is kinds janky.

**Review by sgalsaleh — APPROVED** [review]
